edb/common/struct.py

```diff
@@ -345,8 +344,10 @@ def _check_field_type(self, field, name, value):
             if field.coerce:
                 ftype = field.type[0]
 
-                if issubclass(ftype, (typed.AbstractTypedSequence,
-                                      typed.AbstractTypedSet)):
+                if issubclass(ftype, (checked.CheckedList,
```

Comment:
Is it possible to have a common abc for this, e.g. `checked.Collection`?

Reply:
Lists share a base class, Sets do, too. The reason I went with concrete classes is that the bases are not exported in `__all__` and as such I don't really consider them public API.

Why the preference?

Reply:
Easy to forget to add another concrete class to the check when adding new collection types.

Reply:
I agree, did that then.